pyscripts/r1cs/utils.py

```diff
@@ -0,0 +1,65 @@
+from collections import defaultdict
+
+import json
+
+
+def convert_to_matrix(mat):
+    """
+    :param mat: list(dict{int: int})
+    :return: list(list(int))
+    """
+    res = []
+    size = max(max(k for k in row) for row in mat)
+    for row in mat:
+        r = [0 for _ in range(size + 1)]
+        for k in row:
+            r[k] = row[k]
+        res.append(r)
+    return res
+
+
+def pepper_to_json(file_source_path, filename):
+    res = {
+        'primary_input': [],
+        'aux_input': [],
+        'constraints': []
+    }
+
+    with open(file_source_path + filename + 'primary') as file:
```

Comment:
It isn't necessary (provided one knows how many public inputs are part of the system). I just did it this way because of of the JSON format being separated.